### PR TITLE
Import pages through navigation (part 1)

### PR DESCRIPTION
**Pull request by psrpinto**
> Don't pay attention to the number of commits, this PR basically just creates the `/ui/import/pages` directory, which implements a flow for importing pages. There was a lot of iteration so that's why the high number of commits. Best to just look at the final result.

This PR implements one flow for import pages by selecting the navigation. Later, we can adapt it how we see fit, or we can implement other flows for importing pages that do it differently. This is just a starting point, any feedback you might have can be implemented.

## A note on blueprints

> Blueprints is a temporary name, it should be renamed to something else in the future.

This PR implements importing pages without relying on Blueprints. Blueprints are meant for things that would be crawled, where the user defines the structure of one thing (e.g. blog post), and we than use that to automatically import all other ones.

However, things that don't need to be crawled do not need to rely on a blueprint. Instead, the user can manually import all instances.

This PR implements page import in this manual way, where there is no automatic crawling, so no blueprints are involved. In the future we may decide that we do want to crawl pages, and if so, a new flow can be implemented that relies on blueprints.

## This is only part 1

This PR was getting too large, so I figured I would ship it as is, and there will be a second one that completes the flow. This PR stops after the user selected the pages (see screen recording), the next PR will have the user iteratively import all selected pages, one by one.

## Summary of changes

- Add `LinkField` and a navigation parser. Takes a `<ul>...<ul/>` and returns an array of `LinkField`
- Make it so that the content script has different modes, e.g. "navigation selection" or "generic selection" modes. This was needed because the user clicks on one of the navigation links, but we want to retrieve the parent `<ul>`.
- Implement the flow, under `/ui/import/pages`

## Screen recording

https://github.com/user-attachments/assets/e22a603a-35f7-4cd9-a206-af5b0c413305

### CODE REVIEW

**Comment by psrpinto** [comment]
@akirk @ashfame I'm making progress here on what comes after what is shown in the screen recording (importing each of the pages the user selected). I should have a finished version tomorrow.

My goal is to build a first iteration of the flow here, we can then adapt it however is needed for the demo in other PRs.

File: src/ui/import/pages/ImportPage.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+			selectedPages.length === 0 ||
+			! selectedPages[ pageIndex ] ||
+			selectedPages[ pageIndex ] === ''
+		) {
+			throw Error( `page with index ${ pageIndex } not found` );
+		}
+		setUrl( selectedPages[ pageIndex ] );
+	}, [ session.id, pageIndex, navigate, selectedPages ] );
+
+	return <>{ url }</>;
```
Comment (psrpinto) [inline src/ui/import/pages/ImportPage.tsx:31]: The next PR will implement this component.

Comment (psrpinto) [inline src/ui/import/pages/ImportPage.tsx:31]: The next PR will implement this component.

**Review by ashfame — APPROVED** [review]
Looks good 👍

**Pull request merged**